platform-tests/src/test/java/org/junit/platform/engine/support/descriptor/MethodSourceTests.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.platform.engine.support.descriptor;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.lang.reflect.Method;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.PreconditionViolationException;
+
+class MethodSourceTests {
+
+	@Test
+	void instantiatingWithNullNamesShouldThrowPreconditionViolationException() {
+		assertThrows(PreconditionViolationException.class, () -> new MethodSource(null, null));
+	}
+
+	@Test
+	void instantiatingWithBlankNamesShouldThrowPreconditionViolationException() {
+		assertThrows(PreconditionViolationException.class, () -> new MethodSource("", ""));
+	}
+
+	@Test
+	void instantiationWithNullMethodShouldThrowPreconditionViolationException() {
+		assertThrows(PreconditionViolationException.class, () -> new MethodSource(null));
+	}
+
+	@Test
+	void twoEqualMethodSourceObjectsShouldBeEqual() {
+		assertEquals(new MethodSource("TestClass", "testMethod"), new MethodSource("TestClass", "testMethod"));
+	}
+
+	@Test
+	void twoDifferentMethodSourceObjectsShouldNotBeEqual() {
+		assertNotEquals(new MethodSource("TestClass1", "notEqual"), new MethodSource("TestClass2", "notEqual"));
+	}
+
+	@Test
+	void twoDifferentMethodsInTheSameClassShouldNotBeEqual() {
+		assertNotEquals(new MethodSource("TestClass1", "firstMethod"), new MethodSource("TestClass1", "secondMethod"));
+	}
+
+	@Test
+	void twoSameMethodSourceObjectsShouldHaveSameHashCode() {
+		assertEquals(new MethodSource("TestClass1", "Method1").hashCode(),
+			new MethodSource("TestClass1", "Method1").hashCode());
+	}
+
+	@Test
+	void classNameShouldBeConsistent() throws Exception {
+		Class<?> c = Class.forName("java.lang.String");
```

Comment:
There is no need for reflection here to load the class for `java.lang.String`.

Simply use `String.class`, here and throughout the entire test class.